### PR TITLE
Fixed crashes when no context existed.

### PR DESCRIPTION
**Pull request by StarArawn**

### CODE REVIEW

**Review by MrGVSV — APPROVED** [review]

File: examples/bevy_state.rs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    MainMenu,
+    Options,
+    Play,
+}
+
+fn create_main_menu(mut commands: Commands) {
+    let context = BevyContext::new(|context| {
+        render! {
+            <App>
+                <Text content={"Main Menu".to_string()} size={32.0} />
```
Comment (MrGVSV) [inline examples/bevy_state.rs:22]: Might be worth putting instructions to press space to switch to the next state. Just to flesh it out a bit more.

**Review by MrGVSV — APPROVED** [review]
Looks good! Minor suggestion but not super necessary to merge

File: bevy_kayak_ui/src/lib.rs [hunk 2]
```diff
@@ -36,69 +36,86 @@ pub(crate) fn to_bevy_color(color: &kayak_core::color::Color) -> Color {
...
-    bevy_context: Res<BevyContext>,
-    windows: Res<Windows>,
-    mut cursor_moved_events: EventReader<CursorMoved>,
-    mut mouse_button_input_events: EventReader<MouseButtonInput>,
-    mut char_input_events: EventReader<ReceivedCharacter>,
-    mut keyboard_input_events: EventReader<KeyboardInput>,
-) {
-    let window_size = if let Some(window) = windows.get_primary() {
-        Vec2::new(window.width(), window.height())
+pub fn process_events(world: &mut World) {
```
Comment (StarArawn) [inline bevy_kayak_ui/src/lib.rs:49]: @MrGVSV 
Maybe we should just move process_events into update?

File: bevy_kayak_ui/src/render/unified/font/font_mapping.rs [hunk 3]
```diff
@@ -32,6 +32,11 @@ impl FontMapping {
         }
     }
 
+    pub(crate) fn mark_all_as_new(&mut self) {
```
Comment (StarArawn) [inline bevy_kayak_ui/src/render/unified/font/font_mapping.rs:35]: This just lets us reprocess fonts into the kayak context.

File: examples/bevy_state.rs [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+                <Text content={"Options".to_string()} size={32.0} />
+                <StateSwitcher />
+            </App>
+        }
+    });
+
+    commands.insert_resource(context);
+}
+
+fn create_play_menu(mut commands: Commands) {
```
Comment (StarArawn) [inline examples/bevy_state.rs:75]: The more I think about this the more I dislike it. I think users should instead use scenes here. We should think about how we want to handle that.

File: src/widgets/text.rs [hunk 5]
```diff
@@ -48,30 +50,36 @@ pub fn Text(props: TextProps) {
...
         };
 
-    let render_command = RenderCommand::Text {
-        content: content.clone(),
-        size,
-        parent_size,
-        line_height: line_height.unwrap_or(size * 1.2),
-        font: font_name.clone().unwrap_or("Roboto".into()),
-    };
+    if should_render {
```
Comment (StarArawn) [inline src/widgets/text.rs:67]: This prevents rendering text until all of the conditions are met. I found this useful incases where font has loaded and we don't have the parent layout. It keeps the text from jumping around. :)

File: bevy_kayak_ui/src/lib.rs (hunk 2 above)
Comment (MrGVSV) [reply bevy_kayak_ui/src/lib.rs:49]: Seems reasonable.

File: examples/bevy_state.rs (hunk 4 above)
Comment (MrGVSV) [reply examples/bevy_state.rs:75]: I think it's okay to do for now. We could maybe explore scenes in another PR/RFC?

File: kayak_core/src/context_ref.rs [hunk 6]
```diff
@@ -260,6 +260,14 @@ impl<'a> KayakContextRef<'a> {
             .tree
             .merge(&tree, self.current_id.unwrap_or_default(), changes);
     }
+
+    /// This function is specifically for text rendering which needs to re-render when
+    /// it's parent layout is calculated.
+    pub fn mark_dirty(&mut self, widget_id: &Index) {
```
Comment (MrGVSV) [inline kayak_core/src/context_ref.rs:266]: Why can't we just get `widget_id` from `current_id`?

Comment (StarArawn) [reply kayak_core/src/context_ref.rs:266]: Because I didn't think of it 👍

**Pull request merged**